features/step_definitions/developer_portal/admin/account/payment_details_steps.rb

```diff
@@ -22,7 +22,7 @@
 end
 
 When "the buyer is reviewing their credit card details" do
-  stub_stripe_intent_setup if @buyer.provider_account.payment_gateway_type == :stripe # Only stripe need to be stubbed
+  stub_stripe_intent_setup if @buyer.reload.provider_account.payment_gateway_type == :stripe # Only stripe need to be stubbed
```

Comment:
I am not convinced about this, at all. Problem is the order streps are executed. `@provider` and `@buyer` are first created, then `@provider.payment_gateway_type` is changed and finally `@buyer.provider_account.payment_gateway_type` is read, but it still has the old value 😕.

Reply:
`@buyer.provider_account` and `@provider` might be different Ruby objects depending on how they were created and then operations on them like save, reload, etc. `reload` shouldn't hurt if it makes things working, it is fine with me.